eventBus/eventBus.ts

```diff
@@ -0,0 +1,3 @@
+import Vue from "vue"
+
+export const ModalBus = new Vue()
```

Comment:
Delete this "event bus"
In the modal plugin, add a property on the "modal" plugin and expose it
modal = {
    subscription: new Vue,
     show: function () { this.subscription.$emit } 
 }

Reply:
Implemented. Please check

Reply:
this looks better, remember to delete this file and any references